src/shared/bus-unit-util.c

```diff
@@ -1127,6 +1127,9 @@ static int bus_job_get_service_result(BusWaitForJobs *d, char **result) {
         assert(d->name);
         assert(result);
 
+        if (!endswith(d->name, ".service"))
+                return -EINVAL;
```

Comment:
`Result` is also available on socket, mount, automount, timer, swap, and path units.  (Also scopes, but I suspect that's  not relevant).

Reply:
but the values live in a different namespace, even though they are very similar. Also, we query the one off the Service interface, not the others. That means, if we want to support the other results too, we'd have to add some more code anyway, to use the right dbus interface...

Reply:
(adding support for other unit types should be left for a different PR. For now, we should at least fix that we request the Result property on the wrong interface for non-service units, which this patch does)

Reply:
Sure, that makes sense.